File: spa/src/entities/userInfoClaims.ts
/*
 * User info claims can be returned to the UI or used if the API triggers sending of emails to users
 */
export class UserInfoClaims {

    /*
     * Fields contained
     */
    public givenName: string;
    public familyName: string;
    public email: string;

    /*
     * Construct from input
     */
    public constructor(givenName: string, familyName: string, email: string) {
        this.givenName = givenName;
        this.familyName = familyName;
        this.email = email;
    }
}

File: src/main/java/com/mycompany/sample/host/claims/UserInfoClaims.java
package com.mycompany.sample.host.claims;

import lombok.Getter;

/*
 * User info is returned to the UI or used by the API
 */
public class UserInfoClaims {

    @Getter
    private final String givenName;

    @Getter
    private final String familyName;

    @Getter
    private final String email;

    /*
     * Initialize from input
     */
    public UserInfoClaims(final String givenName, final String familyName, final String email) {
        this.givenName = givenName;
        this.familyName = familyName;
        this.email = email;
    }
}
